pkg/manager/risingwave_controller_manager_impl.go

```diff
@@ -58,9 +58,53 @@ type risingWaveControllerManagerImpl struct {
 	forceUpdateEnabled bool
 }
 
+func getStandaloneStatusUtil(rw *risingwavev1alpha1.RisingWave, logger logr.Logger, readyReplicas, requiredReplicas int32, statefulSetName string) risingwavev1alpha1.ComponentReplicasStatus {
+	status := risingwavev1alpha1.ComponentReplicasStatus{
+		Running: 0,
+		Target:  0,
+	}
+
+	if ptr.Deref(rw.Spec.EnableStandaloneMode, false) {
+		status.Target = 1
+	}
+
+	// we only expect one replica, if standalone mode is enabled
+	if rw.Spec.Components.Standalone != nil {
+		invalidMsg := fmt.Sprintf("warn: Standalone mode enabled, but invalid replica value of %d", rw.Spec.Components.Standalone.Replicas)
+		if ptr.Deref(rw.Spec.EnableStandaloneMode, false) {
+			if rw.Spec.Components.Standalone.Replicas != 1 {
+				logger.Info(invalidMsg)
+			}
+		} else {
+			if rw.Spec.Components.Standalone.Replicas != 0 {
+				logger.Info(invalidMsg)
+			}
+		}
```

Comment:
Running replicas could be 1 in a short period after switching from standalone to cluster, and vice versa. We don't have to warn about that.

Reply:
Fair enough. I originally wanted to have this warning to inform the user about the fact that there may be more than 1 replica configured even though this is standalone.